src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (compareFunction) {
```

Comment:
The check for 'compareFunction' should be done outside of the nested loops. This is a performance issue because you're checking if 'compareFunction' is provided on every iteration, even though it does not change during the sort. Move the check outside the loops to improve performance.